pkg/middleware/redirect_to_https.go

```diff
@@ -26,10 +26,11 @@ func NewRedirectToHTTPS(httpsPort string) alice.Constructor {
 // to the port from the httpsAddress given.
 func redirectToHTTPS(httpsPort string, next http.Handler) http.Handler {
 	return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
-		proto := req.Header.Get("X-Forwarded-Proto")
-		if strings.EqualFold(proto, httpsScheme) || (req.TLS != nil && proto == "") {
-			// Only care about the connection to us being HTTPS if the proto is empty,
-			// otherwise the proto is source of truth
+		proto := requestutil.GetRequestProto(req)
```

Comment:
`X-Forwarded-Proto` is now only trusted to govern the redirect to HTTPS if the ReverseProxy option is set to true.

Reply:
I think the issue before was we pulled the raw `X-Forwarded-Proto` straight from the header with no `ReverseProxy` checks, so client could tamper with input in non-reverse proxy scenarios. Not sure what that gets you in this case... but it is a minor example of us trusting a potential untrustworthy header as canonical.

`GetRequestProto` should return the same `X-Forwarded-Proto` IF it is confirmed behind a Reverse Proxy. Otherwise it returns `req.URL.Scheme` (hence the shift in the final conditional to check `== req.URL.Scheme` instead of `== ""` in the no (or untrusted) X-Forwarded-Proto case.